cli/src/cmd/forge/script/broadcast.rs

```diff
@@ -0,0 +1,260 @@
+use crate::cmd::ScriptSequence;
+
+use ethers::{
+    prelude::{Provider, SignerMiddleware},
+    providers::Middleware,
+    signers::Signer,
+    types::{
+        transaction::eip2718::TypedTransaction, Address, Chain, Eip1559TransactionRequest,
+        TransactionReceipt, TransactionRequest, U256,
+    },
+};
+
+use std::collections::BTreeMap;
+
+use super::*;
+
+impl ScriptArgs {
+    pub async fn send_transactions(
+        &self,
+        deployment_sequence: &mut ScriptSequence,
+    ) -> eyre::Result<()> {
+        // The user wants to actually send the transactions
+        let mut local_wallets = vec![];
+        if let Some(wallets) = self.wallets.private_keys()? {
+            wallets.into_iter().for_each(|wallet| local_wallets.push(wallet));
+        }
+
+        if let Some(wallets) = self.wallets.interactives()? {
+            wallets.into_iter().for_each(|wallet| local_wallets.push(wallet));
+        }
+
+        if let Some(wallets) = self.wallets.mnemonics()? {
+            wallets.into_iter().for_each(|wallet| local_wallets.push(wallet));
+        }
+
+        if let Some(wallets) = self.wallets.keystores()? {
+            wallets.into_iter().for_each(|wallet| local_wallets.push(wallet));
+        }
+
+        // TODO: Add trezor and ledger support (supported in multiwallet, just need to
+        // add derivation + SignerMiddleware creation logic)
+        // foundry/cli/src/opts/mod.rs:110
+        if local_wallets.is_empty() {
+            eyre::bail!("Error accessing local wallet when trying to send onchain transaction, did you set a private key, mnemonic or keystore?")
+        }
+
+        let fork_url = self
+            .evm_opts
+            .fork_url
+            .as_ref()
+            .expect("You must provide an RPC URL (see --fork-url).")
+            .clone();
+        let provider = Provider::try_from(&fork_url).expect("Bad fork provider.");
+
+        let chain = provider.get_chainid().await?.as_u64();
+        let is_legacy =
+            self.legacy || Chain::try_from(chain).map(|x| Chain::is_legacy(&x)).unwrap_or_default();
+        local_wallets =
+            local_wallets.into_iter().map(|wallet| wallet.with_chain_id(chain)).collect();
+
+        // in case of --force-resume, we forgive the first nonce disparity of each from
+        let mut nonce_offset: BTreeMap<Address, U256> = BTreeMap::new();
+
+        // Iterate through transactions, matching the `from` field with the associated
```

Comment:
Would simplify this code a lot by having just 1 wallet and always assume the `from` is the signer?